src/k8s/pkg/k8sd/types/worker.go

```diff
@@ -8,44 +8,32 @@ import (
 
 // InternalWorkerNodeToken encodes information required to join a cluster as a worker node.
 type InternalWorkerNodeToken struct {
-	Token         string   `json:"token"`
+	// Secret is used to verify the join request.
+	// Secret is only valid for a specified worker that was specified upon creation.
+	Secret string `json:"secret"`
+	// JoinAddresses is a list of control-plane addresses that exist in the cluster.
 	JoinAddresses []string `json:"join_addresses"`
 }
 
-// internalWorkerNodeTokenSerializeMagicString is used to validate serialized worker node tokens.
-const internalWorkerNodeTokenSerializeMagicString = "m!!"
-
 var encoding = base64.RawStdEncoding
 
-type serializableWorkerNodeInfo struct {
-	InternalWorkerNodeToken
-	Magic string `json:"_"`
-}
-
 // Encode a worker node token to a base64-encoded string.
```

Comment:
Let's keep this in. If `microcluster` ever changes their tokens to have both `secret` and `join_addresses`, then our check in https://github.com/canonical/k8s-snap/pull/235/files#diff-05ae23d2944a3f9c66ecf2daeccede5edd805e16c27117a47fd45cdd643a8ab4R35 will fail.